torchvision/prototype/models/segmentation/deeplabv3.py

```diff
@@ -72,25 +72,19 @@ class DeepLabV3_MobileNet_V3_Large_Weights(WeightsEnum):
     default = CocoWithVocLabels_V1
 
 
+@handle_legacy_interface(
+    weights=("pretrained", DeepLabV3_ResNet50_Weights.CocoWithVocLabels_V1),
+    weights_backbone=("pretrained_backbone", ResNet50_Weights.ImageNet1K_V1),
+)
 def deeplabv3_resnet50(
+    *,
     weights: Optional[DeepLabV3_ResNet50_Weights] = None,
     progress: bool = True,
     num_classes: Optional[int] = None,
     aux_loss: Optional[bool] = None,
     weights_backbone: Optional[ResNet50_Weights] = None,
-    **kwargs: Any,
```

Comment:
This changes the signature of the method. Though a corner case, it means that now if someone passed an extra parameter as argument it won't be ignored but instead they will get an exception. Note that `kwargs` is currently supported by all model builders so I suggest leaving it in place.

Reply:
> Though a corner case, it means that now if someone passed an extra parameter as argument it won't be ignored but instead they will get an exception.

That is kind of the point. The builder **silently** ignores wrong keyword arguments. Imagine you made a typo

```python
deeplabv3_resnet50(weight=DeepLabV3_ResNet50_Weights.CocoWithVocLabels_V1)
```

and instead of loading some weights, you get a randomly initialized model instead. I don't think this is expected neither a good thing to have. At the very least, we should emit a warning that there were unused keyword arguments.

> Note that `kwargs` is currently supported by all model builders so I suggest leaving it in place.

There are places where this is useful. For example the `**kwargs` of VGG variants like

https://github.com/pytorch/vision/blob/93b26da3623c9134d64f026a234d97caab0675f2/torchvision/prototype/models/vgg.py#L172-L179

will ultimately be resolved in 

https://github.com/pytorch/vision/blob/93b26da3623c9134d64f026a234d97caab0675f2/torchvision/models/vgg.py#L35-L38

So if there is any unknown keyword argument, the class instantiation would fail.

Reply:
@pmeier I understand and I agree with many points you are raising here. My concern is that this is not the right PR to fix this. Fixing it here only resolves the problem for 2 models, leaving the rest affected. Shall we take this on a separate issue?

Reply:
Fair enough :+1: